assembly/src/lib.rs

```diff
@@ -61,3 +43,105 @@ const MAX_U32_ROTATE_VALUE: u8 = 31;
 
 /// The maximum number of bits allowed for the exponent parameter for exponentiation instructions.
 const MAX_EXP_BITS: u8 = 64;
+
+// LIBRARY EXTENSIONS
+// ================================================================================================
+
+#[cfg(feature = "std")]
+pub trait LibraryExt {
```

Comment:
What's the motivation for introducing these extension traits vs. just keeping this functionality as methods on the `Library` struct?

Reply:
Thinking on this further, I think I'm going to move these to `Assembler`, which _will_ be a breaking change (as those functions will disappear from `Library` and `KernelLibrary)` - but requiring the `LibraryExt`/`KernelLibraryExt` traits is already a breaking change anyway (though a trivial one to fix) so I'm not sure it matters really.

Reply:
Agreed - having these on the `Assembler` makes the most sense, I think.

Reply:
Done